### PR TITLE
Mission dev guide

### PR DESCRIPTION
**Pull request by jacoffey3**
Ready for a general content review. I wasn't sure if I should litter links everywhere, but please tag where you think I've left them out. 

I have links stubbed out where I'm planning on putting them, but many just have `#####` instead of the actual link because I just haven't filled them out yet. I'm going through updating that now.

### CODE REVIEW

File: docs/mission-dev/index.rst [hunk 1]
```diff
@@ -1,44 +1,60 @@
...
+Kubos has some mission applications that are open sourced in the repo, and others that we can offer to aid in mission development.
+We've listed the typical necessary applications in the :doc:`mission needs doc. <mission-needs>`
 
-Additionally, :doc:`mission applications <../os-docs/apps/app-guide>` can be easily configured to execute different logic depending on when
-and how they are run. For example, when run on-demand, versus when run at startup.
+3. Update and Recovery
+----------------------
 
-Mission applications are monitored and managed by the :doc:`Kubos applications service <../os-docs/services/app-service>`.
+KubOS uses similar procedures for update and recovery of the rootfs.
```
Comment (cantino) [inline docs/mission-dev/index.rst]: Similar to what?

Comment (jacoffey3) [reply docs/mission-dev/index.rst]: update is similar to recovery

File: docs/mission-dev/index.rst [hunk 2]
```diff
@@ -1,44 +1,60 @@
...
 
-TODO: Typical mission needs:
+Updating KubOS
+______________
 
-- apps (housekeeping, beacon, telemetry)
-- services
-- etc
+The process for updating the rootfs in KubOS can be reviewed here: :doc:`KubOS update process. <#####>`
+The process for updating mission applications can be reviewed here: :doc:`application service guide. <#####>`
```
[the remark this answers is not in the document]
Comment (jacoffey3) [reply docs/mission-dev/index.rst]: Those are 2 different processes. They should be mentioned in the docs that I link to. Hardware services are taken care of by the rootfs (which is mentioned under payload services and should be mentioned under hardware services as well) 

We currently have no specific way to update payload services, and I don't think this doc is the best place to come up with one.

File: docs/mission-dev/index.rst [hunk 3]
```diff
@@ -1,44 +1,60 @@
...
+Kubos offers `SLAs <https://www.kubos.com/kubos/>` to aid in this process.
 
-    mission-needs
-    mission-dev-guide
-    Deployment Application Guide <deployment>
-    flight-ready
+Recovery in KubOS
+_________________
+
+The :doc:`KubOS recovery process <#####>` is defaulted to be configured for a development environment, where you have access to the hardware and are actively developing on it.
```
[the remark this answers is not in the document]
Comment (jacoffey3) [reply docs/mission-dev/index.rst]: Yeah, it should be covered in the mission apps doc, not here.

File: docs/mission-dev/mission-needs.rst [hunk 4]
```diff
@@ -1,10 +1,90 @@
...
+Generally, this should poll all hardware services on a regular basis and pull all available telemetry.
+A 1 minute polling cycle is generally sufficient for most telemetry items.
+
+.. TODO: merge the example app and update to say this: "Kubos provides an `example of this application, <#####>`__ and augmenting it for your mission should be simple given that all hardware services follow the service outline."
+
+Housekeeping
+------------
+
+For each critical piece of the avionics hardware or critical mission aspect, there should be an accompanying housekeeping mission application.
+It is essential to separate these out as much as possible to reduce the complexity of a given application as much as possible, and to reduce possibility of global failure due to an edge case in a single app.
```
Comment (CatGarab) [inline docs/mission-dev/mission-needs.rst]: nit: remove second "as much as possible"

File: docs/mission-dev/mission-needs.rst [hunk 5]
```diff
@@ -1,10 +1,90 @@
...
+- ADCS/GPS
+- Power
+
+These are by no means the *only* housekeeping applications that should be included, but would take care of most of the housekeeping for a given mission.
+
+OBC
+~~~
+
+The OBC housekeeping application should ensure the OBC itself and the critical processes are running smoothly.
+Kubos provides an `example of this application <######>`__, which cleans the telemetry database, checks file system and memory usage, and does a test query to running services.
```
Comment (CatGarab) [inline docs/mission-dev/mission-needs.rst]: nit: "does a test query to running services" -> "issues a test query against all known services"

File: docs/mission-dev/mission-needs.rst [hunk 6]
```diff
@@ -1,10 +1,90 @@
...
+See the :doc:`example application documentation <#####>` to understand more of what it does, and what you might want to do with it or another similar application.
+
+.. TODO: Update with links once the app is merged
+
+ADCS/GPS
+~~~~~~~~
+
+The ADCS/GPS housekeeping app will ensure the hardware itself is behaving normally, checking critical telemetry items to ensure nothing is outside the bounds of normal operation, taking action as appropriate.
```
Comment (CatGarab) [inline docs/mission-dev/mission-needs.rst]: In theory, this app is responsible for periodically fetching/updating position data, right? If so, add a blurb mentioning that.

File: docs/mission-dev/mission-needs.rst [hunk 7]
```diff
@@ -1,10 +1,90 @@
...
+
+Kubos offers housekeeping apps for it's supported ADCS and GPS systems, you can reach out to us on `Slack <https://slack.kubos.co>`__ or through the `website <https://www.kubos.com/kubos/>`__ for more information about these.
+
+Power
+~~~~~
+
+The power management housekeeping application would monitor the Battery and EPS systems, taking critical autonomous recovery action where appropriate.
```
Comment (CatGarab) [inline docs/mission-dev/mission-needs.rst]: nit: "would monitor" -> "should monitor"

File: docs/mission-dev/mission-needs.rst [hunk 8]
```diff
@@ -1,10 +1,90 @@
...
+See the :doc:`example application documentation <#####>` to understand more of what it does, and what you might want to do with it or another similar application.
+
+.. TODO: Update with links once the app is merged
+
+ADCS/GPS
+~~~~~~~~
+
+The ADCS/GPS housekeeping app will ensure the hardware itself is behaving normally, checking critical telemetry items to ensure nothing is outside the bounds of normal operation, taking action as appropriate.
+
+Kubos offers housekeeping apps for it's supported ADCS and GPS systems, you can reach out to us on `Slack <https://slack.kubos.co>`__ or through the `website <https://www.kubos.com/kubos/>`__ for more information about these.
```
Comment (CatGarab) [inline docs/mission-dev/mission-needs.rst]: nits: 
- it's -> its
- Change the comma to a period

File: docs/mission-dev/mission-needs.rst [hunk 9]
```diff
@@ -1,10 +1,90 @@
...
+The deployment application should handle the required sequence during the initial deployment from the launch vehicle.
+We have a :doc:`dedicated guide <deployment>` for the recommended behavior of this application.
+
+Kubos offers configurable deployment applications for customer missions, as well as SLAs for helping develop and/or reviewing your mission's deployment application.
+You can reach out to us on `Slack <https://slack.kubos.co>`__ or through the `website <https://www.kubos.com/kubos/>`__ for more information about these.
+
+Beacon
+------
+
+A beacon application is recommended to generate a compressed subset of telemetry to regularly downlink.
```
Comment (CatGarab) [inline docs/mission-dev/mission-needs.rst]: nit: awkward wording. Maybe:
"The beacon application should regularly downlink a subset of critical telemetry data."

File: docs/mission-dev/mission-needs.rst [hunk 10]
```diff
@@ -1,10 +1,90 @@
...
+
+Kubos offers configurable beacon applications for customer missions, you can reach out to us on `Slack <https://slack.kubos.co>`__ or through the `website <https://www.kubos.com/kubos/>`__ for more information about these.
+
+Operations
+----------
+
+All of the above applications are scoped to the core functionality of a spacecraft.
+They ensure that the spacecraft is functioning properly within operational bounds.
+But, they don't actually *complete* your mission.
+Operations applications should be the functionality that is the objective of the mission itself.
```
Comment (CatGarab) [inline docs/mission-dev/mission-needs.rst]: nit: "should be" -> "should deliver"

File: docs/mission-dev/mission-needs.rst [hunk 11]
```diff
@@ -1,10 +1,90 @@
...
+As a result, they can vary greatly in their functionality, but here are some examples:
```
Comment (CatGarab) [inline docs/mission-dev/mission-needs.rst]: nit: Shorten this to just "For example:"

File: docs/mission-dev/mission-needs.rst [hunk 12]
```diff
@@ -1,10 +1,90 @@
...
+
+- Command the adcs to the appropriate attitude and take a photo of a commanded location
```
Comment (CatGarab) [inline docs/mission-dev/mission-needs.rst]: nit: capitalize ADCS

File: docs/mission-dev/mission-needs.rst [hunk 13]
```diff
@@ -1,10 +1,90 @@
...
+- Collect, store in the telemetry database, and downlink over the communication service specialized sensor data
```
Comment (CatGarab) [inline docs/mission-dev/mission-needs.rst]: nit: awkward. Maybe: "Collect, store, and downlink specialized sensor data"

File: docs/os-docs/services/comms-framework.rst [hunk 14]
```diff
@@ -8,6 +8,8 @@ creating their specific radio's hardware service.
 This framework is used to simplify the process of reading messages from the ground, forwarding them
 to the appropriate internal destination, and then sending properly formatted messages back to the
 ground.
+This framework is meant to be an extension of a hardware service, such that the hardware service GraphQL interface is used for commanding and telemetry to/from the radio hardware itself.
```
Comment (CatGarab) [inline docs/os-docs/services/comms-framework.rst]: nit: Add an extra line to make these two sentences a new paragraph (this intro blurb is just getting a little long)

File: docs/os-docs/services/payload-services.rst [hunk 15]
```diff
@@ -5,6 +5,10 @@ Payload services are essentially hardware services which have been custom design
 for mission payload hardware. They share the same architecture as the hardware
 services, exposing low-level device APIs through a GraphQL interface.
 
+They primarily differ from hardware services in how they are integrated into the final system.
+Hardware services each have an accompanying Buildroot package for including them at build time, so they are part of the rootfs and it's associated recovery process.
```
Comment (CatGarab) [inline docs/os-docs/services/payload-services.rst]: nit: Maybe reword this sentence:
Hardware services each have an accompanying Buildroot package, allowing them to be built into the base operating system image. This allows them to be covered by the operating system's recovery process.

File: docs/mission-dev/index.rst [hunk 16]
```diff
@@ -1,44 +1,60 @@
...
 
-    mission-needs
-    mission-dev-guide
-    Deployment Application Guide <deployment>
-    flight-ready
+Recovery in KubOS
+_________________
+
+The :doc:`KubOS recovery process <../os-docs/linux-docs/kubos-linux-recovery>` is defaulted to be configured for a development environment, where you have access to the hardware and are actively developing on it.
+As a result, it **requires** augmentation before being flight ready, as it is not initially configured for desired on-orbit behavior.
```
[the remark this answers is not in the document]
Comment (jacoffey3) [reply docs/mission-dev/index.rst]: I think this should actually be called out in that recovery doc as well. But I'll include it here.

File: docs/mission-dev/index.rst [hunk 17]
```diff
@@ -1,44 +1,60 @@
...
+2. Mission Applications
+-----------------------
 
-KubOS is built to support multiple mission application files existing and running on a system concurrently.
-This allows customers to break their logic into specific, siloed portions which can be updated or replaced piecemeal
-as issues arise or as new features are added.
+In KubOS, services only implement the minimal amount of decision making necessary to accomplish their given task.
+Instead, we rely on :ref:`mission applications <app-docs>` to handle the decision making for the mission.
+Kubos has some mission applications that are open sourced in the repo, and others that we can offer to aid in mission development.
+We've listed the typical necessary applications in the :doc:`mission needs doc. <mission-needs>`
```
[the remark this answers is not in the document]
Comment (jacoffey3) [reply docs/mission-dev/index.rst]: I'm not sure exactly how to do that...can you do the suggest change thing?

File: docs/mission-dev/index.rst (hunk 16 above)
[the remark this answers is not in the document]
Comment (CatGarab) [reply docs/mission-dev/index.rst]: Ooh, good point

File: docs/mission-dev/index.rst [hunk 18]
```diff
@@ -1,44 +1,63 @@
...
+-----------------------
 
-KubOS is built to support multiple mission application files existing and running on a system concurrently.
-This allows customers to break their logic into specific, siloed portions which can be updated or replaced piecemeal
-as issues arise or as new features are added.
+In KubOS, services are used primarily to expose functionality of underlying hardware.
+They should be expected to only perform a minimal amount of decision-making (for example, kicking a watchdog at a pre-defined interval)
+Instead, we rely on :ref:`mission applications <app-docs>` to handle the decision making for the mission.
+Kubos has some mission applications that are open sourced in the repo, and others that we can offer to aid in mission development.
+We've listed the typical necessary applications in the :doc:`mission needs doc. <mission-needs>`
```
Comment (CatGarab) [inline docs/mission-dev/index.rst]:
```suggestion
We've listed the typical necessary applications in the :doc:`mission needs doc. <mission-needs>`

.. toctree::
    :hidden:
    
    mission-needs
    deployment
```

File: docs/mission-dev/index.rst [hunk 19]
```diff
@@ -1,44 +1,63 @@
...
+In addition, we also recommend doing several test cycles forcing the recovery to take place under various conditions, verifying that your mission specific code is effectively integrated into the recovery process.
+
+Kubos offers `SLAs <https://www.kubos.com/kubos/>`__ to aid in the augmentation of the process and/or auditing your mission's recovery process.
+
+4. Flight Readiness
+-------------------
+
+You have all the hardware integrated, you have all the required mission applications, and you've tested everything to your heart's content...you're ready for launch.
+We've created a checklist for the :doc:`likely steps <flight-ready>` to the prepare the software to be launch-ready.
+Please note that this is only a recommended list of steps, and will definitely change depending on your launch provider, mission requirements, and hardware configuration.
```
Comment (CatGarab) [inline docs/mission-dev/index.rst]:
```suggestion
Please note that this is only a recommended list of steps, and will definitely change depending on your launch provider, mission requirements, and hardware configuration.

.. toctree::
    :hidden:
    
    flight-ready
```

File: docs/mission-dev/index.rst [hunk 20]
```diff
@@ -1,44 +1,63 @@
...
-- Payload monitoring
-- :doc:`Deployment <deployment>`
+2. Mission Applications
+-----------------------
 
-KubOS is built to support multiple mission application files existing and running on a system concurrently.
-This allows customers to break their logic into specific, siloed portions which can be updated or replaced piecemeal
-as issues arise or as new features are added.
+In KubOS, services are used primarily to expose functionality of underlying hardware.
+They should be expected to only perform a minimal amount of decision-making (for example, kicking a watchdog at a pre-defined interval)
```
Comment (CatGarab) [inline docs/mission-dev/index.rst]: nit: missing a period at the end of the sentence

File: docs/mission-dev/mission-needs.rst [hunk 21]
```diff
@@ -1,10 +1,93 @@
...
+
+The power management housekeeping application should monitor the Battery and EPS systems, taking critical autonomous recovery action where appropriate.
+Some examples of triggers and actions:
+
+- Shutting off non-essential hardware when battery reaches critically low status
+- No charging detected for X time period
+- Cancelling operations and going into power generation state
+- Battery temperature monitoring
+
+Kubos offers housekeeping apps for it's supported power systems, you can reach out to us on `Slack <https://slack.kubos.co>`__ or through the `website <https://www.kubos.com/kubos/>`__ for more information about these.
```
Comment (CatGarab) [inline docs/mission-dev/mission-needs.rst]: nits: 
- it's -> its
- change the comma to a period

File: docs/mission-dev/mission-needs.rst [hunk 22]
```diff
@@ -1,10 +1,93 @@
...
+Kubos offers configurable deployment applications for customer missions, as well as SLAs for helping develop and/or reviewing your mission's deployment application.
+You can reach out to us on `Slack <https://slack.kubos.co>`__ or through the `website <https://www.kubos.com/kubos/>`__ for more information about these.
+
+Beacon
+------
+
+The beacon application should regularly downlink a subset of critical telemetry data via the communication service.
+This telemetry should be the absolute minimum required to assess overall system health.
+
+Kubos offers configurable beacon applications for customer missions, you can reach out to us on `Slack <https://slack.kubos.co>`__ or through the `website <https://www.kubos.com/kubos/>`__ for more information about these.
```
Comment (CatGarab) [inline docs/mission-dev/mission-needs.rst]: nit: comma -> period
